core/crypto.go

```diff
@@ -84,16 +84,25 @@ func Hash(b []byte) []byte {
 
 // Scramble fills a given buffer with cryptographically-secure random bytes.
 func Scramble(buf []byte) {
+	// TODO: This should call some assembly routines like Wipe does, for the same
```

Comment:
What would those assembly routines do? The only option is probably to implement system calls but that's highly platform specific and I'm not sure why some assembly would be less likely to be optimized out.

The compiler likely has some heuristic based on side-effects and some assembly doing the same thing as a standard call would be no different in this regard.

Perhaps there's some way of tricking the compiler into thinking that there's a side effect, by mutating some global or accessing the slice right after.

Reply:
The hope was that some no-op trick could be done like here: https://boringssl.googlesource.com/boringssl/+/787b26cc51dc5f4b3f423e045bede996038bddd8/crypto/mem.c#156

An alternative would be something like this: https://github.com/golang/go/issues/33325

Reply:
Referring to discussion in the proposal you opened, what do you think about adding `KeepAlive` calls to these functions instead?

Reply:
Yeah, appending a KeepAlive call is probably better because of maintainability.